pkg/machinery/resources/network/link_refresh.go

```diff
@@ -19,57 +21,46 @@ const LinkRefreshType = resource.Type("LinkRefreshes.net.talos.dev")
 //
 // Whenever Wireguard interface is updated, LinkRefresh resource is modified to trigger a reconcile
 // loop in the LinkStatusController.
-type LinkRefresh struct {
-	md   resource.Metadata
-	spec LinkRefreshSpec
-}
+type LinkRefresh = typed.Resource[LinkRefreshSpec, LinkRefreshRD]
 
 // LinkRefreshSpec describes status of rendered secrets.
 type LinkRefreshSpec struct {
-	Generation int `yaml:"generation"`
+	Generation *int `yaml:"generation"`
```

Comment:
I might be missing something, but I thought `TypedSpec()` returns a pointer to the spec? 

so it should be fine to do `(*LinkRefreshSpec) Bump` ?

Reply:
Not really. `TypedSpec()` indeed returns a pointer, but we still need to defined method with non-pointer receiver, since generic constraints expects method to have a non-pointer receiver. Basically `(*LinkRefreshSpec) Bump` does not compile.

Reply:
Sounds like a limitation to me, why should it care... 

I wonder if two values of LinkRefresh resource are still equal after DeepCopy? I hope they are, as it's using reflect.DeepEqual behind the scenes

Reply:
I tried to explain the limitation here: https://go.dev/play/p/NObWzQao8cR